src/main/java/io/tus/java/client/persistence/TusURLPersistentStore.java

```diff
@@ -0,0 +1,182 @@
+package io.tus.java.client.persistence;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+import org.apache.commons.io.FileUtils;
+
+import io.tus.java.client.TusURLStore;
+
+public class TusURLPersistentStore implements TusURLStore {
+	
+	private static volatile TusUploadFileSet filesWithFingerprints;

```

Comment:
Why is this property static? Does this mean that the user can only effectively use one TusURLPersistentStore?

Reply:
Honestly I don't remember why is it static. However I didn't recognize any use case for more than one TusUploadFileSet.